odxtools/odxtypes.py

```diff
@@ -157,7 +157,7 @@ def compare_odx_values(a: AtomicOdxType, b: AtomicOdxType) -> int:
 
 # format specifiers for the data type using the bitstruct module
 _BITSTRUCT_FORMAT_LETTER_MAP__ = {
-    "A_INT32": "s",
+    "A_INT32": "u",  # the sign must be handled separately
```

Comment:
Does the new code, handle the case base_type_encoding is None and A_INT32 used?

Reply:
yes. It needs to deal with negative numbers manually, though. (before this, negative numbers always used the two-complement.)

Reply:
in case your question was what happens if the base type is not specified: specifying a base type is mandatory for all occasions where a base type encoding can be specified. (i.e., for DIAG-CODED-TYPE)